docs/introduction/why-rtk-is-redux-today.md

```diff
@@ -0,0 +1,220 @@
+
+Additionally, store setup usually required a few dozen lines of setting up commonly used middleware like thunks, hooking up the Redux DevTools Extension support
+
+### What Does Redux Toolkit Do?
+
+While these _were_ the patterns originally shown in the Redux docs, they unfortunately require a lot of very verbose and repetitive code. Most of this boilerplate isn't _necessary_ to use Redux. On top of that, the boilerplate-y code lead to more opportunities to make mistakes.
+
+**We specifically created Redux Toolkit to eliminate the "boilerplate" from hand-written Redux logic, prevent common mistakes, and provide APIs that simplify standard Redux tasks**.
+
+Redux Toolkit starts with two key APIs that simplify the most common things you do in every Redux app:
+
+- `configureStore` sets up a well-configured Redux store with a single function call, including combining reducers, adding the thunk middleware, and setting up the Redux DevTools integration. It also is easier to configure than `createStore`, because it takes named options parameters.
+- `createSlice` lets you write reducers that use [the Immer library](https://immerjs.github.io/immer/) to enable writing "mutating" update syntax that becomes safe and correct immutable updates, with no spreads needed. It also automatically generates action creator functions for each reducer, and generates action type strings internally based on your reducer's names. Finally, it works great with TypeScript.
+
+That means that the code _you_ write can be drastically simpler. For example, that same todos reducer could just be:
+
+```js title="features/todos/todosSlice.js"
+import { createSlice } from '@reduxjs/toolkit'
+
+const todosSlice = createSlice({
+  name: 'todos',
+  initialState: [],
+  reducers: {
+    todoAdded(state, action) {
+      state.push({
+        id: action.payload.id,
+        text: action.payload.text,
+        completed: false
+      })
+    },
+    todoToggled(state, action) {
+      const todo = state.find(todo => todo.id === action.payload)
+      todo.completed = !todo.completed
+    }
+  }
+})
+
+export const { todoAdded, todoToggled } = todosSlice.actions
+export default todosSlice.reducer
+```
+
+and store setup can be simplified down to:
+
+```js title="app/store.js"
+import { configureStore } from '@reduxjs/toolkit'
+import todosReducer from '../features/todos/todosSlice'
+import filtersReducer from '../features/filters/filtersSlice'
+
+export const store = configureStore({
+  reducer: {
+    todos: todosReducer,
+    filters: filtersReducer
+  }
+})
+
+// The store has been created with these options:
+// - The slice reducers were automatically passed to combineReducers()
+// - The redux-thunk middleware was automatically added
+// - Dev-mode middleware was added to catch accidental mutations
```

Comment:
If that doesn't come later anyways, can we add a bit more info on the dev-mode middleware?